src/util.js

```diff
@@ -103,176 +66,13 @@ export const getConnection = defer(
-    return hosts[id];
-  }
-
-  throw new Error("no hosts found");
-}
-
-// ==================================================================
-
-export async function getNetworkId(xo) {
-  const config = await getConfig();
-  const networks = xo.objects.indexes.type.network;
-  const network = find(networks, { name_label: config.network });
-  return network.id;
-}
-
-// ==================================================================
-
-export async function getVmXoTestPvId(xo) {
-  const config = await getConfig();
-  const vms = xo.objects.indexes.type.VM;
-  const vm = find(vms, { name_label: config.pvVm });
-  return vm.id;
-}
-
-export async function getVmToMigrateId(xo) {
-  const config = await getConfig();
-  const vms = xo.objects.indexes.type.VM;
-  const vm = find(vms, { name_label: config.vmToMigrate });
-  return vm.id;
-}
-
-// ==================================================================
-
-export async function getSrId(xo) {
-  const host = getOneHost(xo);
-  const pool = await xo.getOrWaitObject(host.$poolId);
-  return pool.default_SR;
-}
-
-// ==================================================================
-
-export async function jobTest(xo) {
-  const vmId = await getVmXoTestPvId(xo);
-  const jobId = await xo.call("job.create", {
-    job: {
-      type: "call",
-      key: "snapshot",
-      method: "vm.snapshot",
-      paramsVector: {
-        type: "cross product",
-        items: [
-          {
-            type: "set",
-            values: [
-              {
-                id: vmId,
-                name: "snapshot",
-              },
-            ],
-          },
-        ],
-      },
-    },
-  });
-  return jobId;
-}
-
-export async function scheduleTest(xo, jobId) {
-  const schedule = await xo.call("schedule.create", {
-    jobId: jobId,
-    cron: "* * * * * *",
-    enabled: false,
-  });
-  return schedule;
-}
-
-export async function getSchedule(xo, id) {
-  const schedule = xo.call("schedule.get", { id: id });
-  return schedule;
-}
-
-// ==================================================================
-
-export function deepDelete(obj, path) {
-  const lastIndex = path.length - 1;
-  for (let i = 0; i < lastIndex; i++) {
-    obj = obj[path[i]];
-
-    if (typeof obj !== "object" || obj === null) {
-      return;
-    }
-  }
-  delete obj[path[lastIndex]];
-}
-
-export function almostEqual(actual, expected, ignoredAttributes) {
-  actual = cloneDeep(actual);
-  expected = cloneDeep(expected);
-  forEach(ignoredAttributes, ignoredAttribute => {
-    deepDelete(actual, ignoredAttribute.split("."));
-    deepDelete(expected, ignoredAttribute.split("."));
-  });
-  expect(actual).to.be.eql(expected);
-}
-
-export async function waitObjectState(xo, id, predicate) {
-  let obj = xo.objects.all[id];
-  while (true) {
-    try {
-      await predicate(obj);
-      return;
-    } catch (_) {}
-    // If failed, wait for next object state/update and retry.
-    obj = await xo.waitObject(id);
-  }
-}
```

Comment:
Why all these changes?

Reply:
I recommande to @heafalan to create a new file which only contains needed functions. But sure, this changement should not be in this PR.

Reply:
To keep only the functions we need.

Reply:
What you could do is simply create more specific modules and progressively shrink `utils` until it's no longer necessary.

Reply:
Good idea :+1: